src/main/python/camdkit/framework.py

```diff
@@ -7,22 +7,188 @@
+class Rotator3:
+  "3 doubles pan, tilt, roll to encode - for example - a camera rotation."
+  pan: typing.Optional[numbers.Real] = 0.0
+  tilt: typing.Optional[numbers.Real] = 0.0
+  roll: typing.Optional[numbers.Real] = 0.0
+
+@dataclasses.dataclass
+class Transform:
+  "A translation, rotation and scale. 'name' and 'parent' fields enable geometry chains"
+  translation: Vector3
+  rotation: Rotator3
+  scale: typing.Optional[Vector3] = None
+  name: typing.Optional[str] = None
+  parent: typing.Optional[str] = None
+
+@dataclasses.dataclass
+class Encoders:
+  "Normalised FIZ encoder values"
+  focus: typing.Optional[float] = None
+  iris: typing.Optional[float] = None
+  zoom: typing.Optional[float] = None
+
+@dataclasses.dataclass
+class ExposureFalloff:
+  "Coefficients for the calculation of exposure fall-off"
+  a1: float
+  a2: typing.Optional[float] = None
+  a3: typing.Optional[float] = None
+
+@dataclasses.dataclass
+class Distortion:
+  "Coefficients for the calculation of radial and (optionally) tangential lens distortion"
+  radial: typing.Tuple[float]
+  tangential: typing.Optional[typing.Tuple[float]] = None
+
+@dataclasses.dataclass
+class PerspectiveShift:
+  "Shift in x and y of the centre of projection of the virtual camera"
+  Cx: float
+  Cy: float
+  
+@dataclasses.dataclass
+class CentreShift:
+  "Shift in x and y of the centre of distortion of the virtual camera"
+  cx: float
+  cy: float
+
+@dataclasses.dataclass
+class GlobalPosition:
+  "Global ENU and geodetic coordinates"
+  E: float
+  N: float
+  U: float
+  lat0: float
+  lon0: float
+  h0: float
+
+@dataclasses.dataclass
+class Timestamp:
+  """
+  A 48-bit integer representing seconds, and a 32-bit integer representing nanoseconds, and an
+  optional 32-bit integer representing attoseconds elapsed since 00:00 January 1st 1970 epoch.
+  Reference: https://datatracker.ietf.org/doc/html/rfc8877
+  """
+  seconds: int
+  nanoseconds: int
+  attoseconds: typing.Optional[int] = None
+
+class BaseEnum(Enum):
+  "Base class for enumerations"
+
+  def __str__(self):
+    return self.value
+
+
+class SynchronizationSourceEnum(BaseEnum):
+  GENLOCK = "genlock"
+  VIDEO_IN = "videoIn"
+  PTP = "ptp"
+  NTP = "ntp"
+
+@dataclasses.dataclass
+class SynchronizationOffsets:
+  """
+  Offsets in seconds between sync and sample. Critical for e.g. frame remapping, or when using
+  different data sources for position/rotation and lens encoding
+  """
+  translation: typing.Optional[float] = None		
+  rotation: typing.Optional[float] = None		
+  encoders: typing.Optional[float] = None
+
+  def validate(self):
+    return all([isinstance(self.translation, float), 
+                isinstance(self.rotation, float), isinstance(self.encoders, float)])
+  
+  @staticmethod
+  def to_json(value: typing.Any) -> typing.Any:
+    return dataclasses.asdict(value)
+
+class TimingModeEnum(BaseEnum):
+  INTERNAL = "internal"
+  EXTERNAL = "external"
+
+class TimecodeFormat(BaseEnum):
```

Comment:
The enumeration is missing frame rates.

In order to fully parameterize timecode, you need two parameters: frame rate and whether the timecode is drop frame or non drop frame.